components/org.wso2.carbon.data.provider/src/main/java/org/wso2/carbon/data/provider/endpoint/DataProviderEndPoint.java

```diff
@@ -112,16 +113,51 @@ public void onMessage(String message, Session session) {
                         dataProviderConfigRoot.getDataProviderConfiguration()).start();
                 getDataProviderHelper().addDataProviderToSessionMap(session.getId(), dataProviderConfigRoot.getTopic(),
                         dataProvider);
+                if (dataProvider instanceof SiddhiProvider) {
+                    SiddhiProvider siddhiProvider = (SiddhiProvider)dataProvider;
+                    if(siddhiProvider.getSiddhiDataProviderConfig().isPaginationEnable()){
+                        siddhiProvider.publishWithPagination(dataProviderConfigRoot.getDataProviderConfiguration(),
+                                dataProviderConfigRoot.getTopic(), session.getId());
+                    }
+                }
             } else if (dataProviderConfigRoot.getAction().equalsIgnoreCase(DataProviderConfigRoot.Types.UNSUBSCRIBE
                     .toString())) {
                 getDataProviderHelper().removeTopicIfExist(session.getId(), dataProviderConfigRoot.getTopic());
+            } else if (dataProviderConfigRoot.getAction().equalsIgnoreCase(DataProviderConfigRoot.Types.POLLING
```

Comment:
There are  5 levels of nested if condition which is not the best practice. Please try to invert the if condition and reduce the nested if.

Reply:
Fixed in dd7e38e